### ISSUE TITLE
clangd shipped with sdk does not find some headers

### ISSUE THREAD

**Issue by atn34** [llvm-mos/llvm-mos#203]
In particular, clangd does not find `<string.h>`. It also doesn't find the dodo-specific `<api.h>` (which would be awesome!), but I can work around that more easily.

```
$ cat build/compile_commands.json 
[
{
  "directory": "/Users/anoyes/workspace/dodo-games/build",
  "command": "/Users/anoyes/llvm-mos/bin/mos-dodo-clang++ -I/Users/anoyes/workspace/dodo-games/menu -I/Users/anoyes/workspace/dodo-games/fixed_point -isystem /Users/anoyes/workspace/dodo-games/api -Os -DNDEBUG -Oz -std=gnu++17 -o CMakeFiles/bounce.dir/bounce.cpp.obj -c /Users/anoyes/workspace/dodo-games/bounce.cpp",
  "file": "/Users/anoyes/workspace/dodo-games/bounce.cpp"
},
{
  "directory": "/Users/anoyes/workspace/dodo-games/build",
  "command": "/Users/anoyes/llvm-mos/bin/mos-dodo-clang++ -I/Users/anoyes/workspace/dodo-games/menu -isystem /Users/anoyes/workspace/dodo-games/api -Os -DNDEBUG -Oz -std=gnu++17 -o menu/CMakeFiles/menu.dir/menu.cpp.obj -c /Users/anoyes/workspace/dodo-games/menu/menu.cpp",
  "file": "/Users/anoyes/workspace/dodo-games/menu/menu.cpp"
},
{
  "directory": "/Users/anoyes/workspace/dodo-games/build",
  "command": "/Users/anoyes/llvm-mos/bin/mos-dodo-clang++ -I/Users/anoyes/workspace/dodo-games/fixed_point -isystem /Users/anoyes/workspace/dodo-games/api -Os -DNDEBUG -Oz -std=gnu++17 -o fixed_point/CMakeFiles/fixed_point.dir/fixed_point.cpp.obj -c /Users/anoyes/workspace/dodo-games/fixed_point/fixed_point.cpp",
  "file": "/Users/anoyes/workspace/dodo-games/fixed_point/fixed_point.cpp"
}
]
```

**Comment by mysterymath** [comment]
As far as I can tell, everything was working as intended.

The trick is that the flag `--query-driver` actually accepts a wildcard, not a specific driver to query. I was thinking that you had to set it manually to the specific driver you wanted to query for install paths, which would be onerous for the SDK. But you can actually just set it to `/path/to/sdk/bin/*clang*`, and it will take the specific driver to query from the `compile_commands.json`. The flag just authorizes clangd to do so, for some security reason or another.

With that, I was able to get VSCode to include target-specific files. Closing this as WAI; please reopen if this doesn't end up working.

I've added a full guide on how to set up clangd here: https://llvm-mos.org/wiki/Clangd

**Comment by atn34** [comment]
Awesome! I've got it working with
```
$ cat .vscode/settings.json
{
    "clangd.path": "/Users/anoyes/llvm-mos/bin/clangd",
    "clangd.arguments": [
        "--query-driver=/Users/anoyes/llvm-mos/bin/*clang*"
    ]
}
```

RE https://llvm-mos.org/wiki/Clangd - it seems that the wiki mentions `--query-compiler` but I only got it to work with `--query-driver`. Also it didn't work with double quotes in the argument. I went ahead and updated the wiki based on that. I was unsure if I should remove the quotes, since they'd be necessary if a shell were interpreting the command, and decided that I prefer no quotes since I expect editor configuration to be a list of args rather than a shell command. At least that's how vscode's configuration works.

Thanks!
